datalogistik/util.py

```diff
@@ -284,16 +287,108 @@ def schema_to_dict(schema):
     return field_dict
 
 
+def convert_arrow_alias(type_name):
+    aliases = {
+        "bool": "bool_",
+        "halffloat": "float16",
+        "float": "float32",
+        "double": "float64",
+        "decimal": "decimal128",
+    }
+    for (alias, func_name) in aliases.items():
+        if type_name == alias:
+            return func_name
+    # no alias was found
+    return type_name
+
+
+# Create an instance of the pyarrow datatype with the given name
+def arrow_type_function_lookup(function_name):
+    if isinstance(function_name, str):
+        function_name = convert_arrow_alias(function_name)
+        pa_type_func = getattr(pa, function_name)
+        return pa_type_func
+
+    # The argument was not a pyarrow type (maybe a nested structure?)
+    return None
+
+
+# Convert a given item (string or dict) to the corresponding Arrow datatype
+def arrow_type_from_json(input_type):
+    arrow_nested_types = {
+        "list_",
+        "large_list",
+        "map_",
+        "struct",
+        "dictionary",
+        # Could be useful for the user to have control over nullability
+        "field",
+    }
+
+    # In case the type is a simple string
+    if isinstance(input_type, str):
+        if input_type in arrow_nested_types:
+            msg = "Nested types in schema not supported yet"
+            log.error(msg)
+            raise ValueError(msg)
+        return arrow_type_function_lookup(input_type)()
+
+    # Alternatively, a type can be encoded as a name:value pair
+    if not input_type.get("type_name"):
+        msg = "Schema field type 'type_name' missing"
+        log.error(msg)
+        raise ValueError(msg)
+
+    type_name = input_type.get("type_name")
+    args = input_type.get("arguments")
+    if type_name in arrow_nested_types:
+        msg = "Nested types in schema not supported yet"
+        log.error(msg)
+        raise ValueError(msg)
+
+    if args is None:
+        return arrow_type_function_lookup(type_name)()
+    if isinstance(args, Mapping):
+        return arrow_type_function_lookup(type_name)(**args)
+    elif isinstance(args, list):
+        log.debug(f"args {args}")
+        return arrow_type_function_lookup(type_name)(*args)
+    else:  # args is probably a single value
+        return arrow_type_function_lookup(type_name)(args)
+
+
+# Convert the given dict to a pyarrow.schema
+def get_arrow_schema(input_schema):
+    log.debug("Converting schema to pyarrow.schema...")
+    if input_schema is None:
+        return None
```

Comment:
nit: maybe log after we've checked for null input?
```suggestion
    if input_schema is None:
        return None
    log.debug("Converting schema to pyarrow.schema...")
```

Reply:
Done